<p>Many pipelines involving annotation/assembly comparison involve Blast (<a href="https://blast.ncbi.nlm.nih.gov/Blast.cgi" class="uri">https://blast.ncbi.nlm.nih.gov/Blast.cgi</a>). Several Blast versions are available as modules, for example:</p>
<ul>
<li><strong>blast/2.12.0+</strong>, etc. : the Blast+ suites (<code>blastp</code>, <code>tblastn</code>, etc.), <strong>recommended</strong></li>
<li><strong>blast/2.2.26</strong>, etc. : 'legacy' Blast (<code>blastall</code>, <code>megablast</code>, etc)</li>
</ul>
<p>Use <code>module spider blast</code> to see available versions. As for all bioinformatics tools at Uppmax, <code>module load bioinfo-tools</code> is required before the blast modules are available.</p>
<p>Uppmax maintains local copies of many Blast databases, including many available at NCBI:</p>
<ul>
<li><a href="ftp://ftp.ncbi.nih.gov/blast/db/README" class="uri">ftp://ftp.ncbi.nih.gov/blast/db/README</a></li>
<li><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/documents/blastdb.html" class="uri">ftp://ftp.ncbi.nlm.nih.gov/blast/documents/blastdb.html</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/books/NBK62345/" class="uri">https://www.ncbi.nlm.nih.gov/books/NBK62345/</a></li>
<li><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/" class="uri">https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/sars-cov-2/" class="uri">https://www.ncbi.nlm.nih.gov/sars-cov-2/</a></li>
<li><a href="https://www.ncbi.nlm.nih.gov/refseq/refseq_select/" class="uri">https://www.ncbi.nlm.nih.gov/refseq/refseq_select/</a></li>
<li><a href="https://blast.ncbi.nlm.nih.gov/smartblast/smartBlast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs#searchSets" class="uri">https://blast.ncbi.nlm.nih.gov/smartblast/smartBlast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs#searchSets</a></li>
</ul>
<p>as well as several UniProt databases (<a href="http://www.uniprot.org/" class="uri">http://www.uniprot.org/</a>).</p>
<p>Note that:</p>
<ul>
<li>The local UPPMAX copies are found at <code>/sw/data/blast_databases</code></li>
<li>Doing <code>module load blast_databases</code> sets the environment variable <code>BLASTDB</code> to this directory; this is loaded as a prerequisite when loading any blast modules</li>
<li>New versions are installed <strong>the first day of each month at 00.01</strong> from local copies updated <strong>the 28th of the previous month beginning at 00.01</strong></li>
<li>When new versions are installed, the directory containing the previous versions is renamed to <code>blast_databases_old</code></li>
<li><code>blast_databases_old</code> is deleted the second data of each month at 00.01</li>
</ul>
<p>These databases use the "v5" format, which includes rich taxonomic infromation with sequences, and will only work with the Blast tools from the module blast/2.8.0+ and later. Earlier module versions can still be used, but you will need to provide/build your own databases. NCBI no longer updates databases with the older "v4" databases as of February 2020, and they have been deleted from UPPMAX. The final updates of these databases (again, as of this writing nearly two years old) are available from NCBI over FTP at <a href="ftp://ftp.ncbi.nlm.nih.gov/blast/db/v4" class="uri">ftp://ftp.ncbi.nlm.nih.gov/blast/db/v4</a>.</p>
<p>Each NCBI-hosted database also includes a JSON file containing additional medatadata for that particular database. These are found in <code>/sw/data/blast_databases/</code> and are named <em>databasename</em>*.json. The exact name varies based on the format of the database. For example, the contents of the JSON file for the <code>nr</code> database can be see by running</p>
<pre><code>cat /sw/data/blast_databases/nr*.json</code></pre>
<p>The Blast databases available at UPPMAX are:</p>
<table>
<colgroup>
<col style="width: 20%" />
<col style="width: 20%" />
<col style="width: 24%" />
<col style="width: 36%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Name</strong></th>
<th><strong>Type</strong></th>
<th><strong>Source</strong></th>
<th><strong>Notes</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>16S_ribosomal_RNA</strong></td>
<td>nucleotide</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/">NCBI</a></td>
<td>16S ribosomal RNA (Bacteria and Archaea type strains)</td>
</tr>
<tr class="even">
<td><strong>18S_fungal_sequences</strong></td>
<td>nucleotide</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/">NCBI</a></td>
<td>18S ribosomal RNA sequences (SSU) from Fungi type and reference material (BioProject PRJNA39195)</td>
</tr>
<tr class="odd">
<td><strong>28S_fungal_sequences</strong></td>
<td>nucleotide</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/">NCBI</a></td>
<td>28S ribosomal RNA sequences (LSU) from Fungi type and reference material (BioProject PRJNA51803)</td>
</tr>
<tr class="even">
<td><strong>Betacoronavirus</strong></td>
<td>nucleotide</td>
<td><a href="https://www.ncbi.nlm.nih.gov/sars-cov-2/">NCBI</a></td>
<td>Betacoronavirus nucleotide sequences</td>
</tr>
<tr class="odd">
<td><strong>cdd_delta</strong></td>
<td>protein</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Conserved domain database for use with <code>delta-blast</code></td>
</tr>
<tr class="even">
<td><strong>env_nr</strong></td>
<td>protein</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Protein sequences for metagenomes (EXCLUDED from nr)</td>
</tr>
<tr class="odd">
<td><strong>env_nt</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Nucleotide sequences for metagenomes</td>
</tr>
<tr class="even">
<td><strong>human_genome</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/documents/blastdb.html">NCBI</a></td>
<td>Current RefSeq human genome assembly with various database masking</td>
</tr>
<tr class="odd">
<td><strong>ITS_eukaryote_sequences</strong></td>
<td>nucleotide</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/">NCBI</a></td>
<td>Internal transcribed spacer region (ITS) for eukaryotic sequences</td>
</tr>
<tr class="even">
<td><strong>ITS_RefSeq_Fungi</strong></td>
<td>nucleotide</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/">NCBI</a></td>
<td>Internal transcribed spacer region (ITS) from Fungi type and reference material (BioProject PRJNA177353)</td>
</tr>
<tr class="odd">
<td><strong>landmark</strong></td>
<td>protein</td>
<td><a href="https://blast.ncbi.nlm.nih.gov/smartblast/smartBlast.cgi?CMD=Web&amp;PAGE_TYPE=BlastDocs#searchSets">NCBI</a></td>
<td>Proteomes of 27 model organisms. The landmark database includes complete proteomes from a few selected representative genomes spanning a wide taxonomic range, the main database used by the SmartBLAST services.</td>
</tr>
<tr class="even">
<td><strong>LSU_eukaryote_rRNA</strong></td>
<td>nucleotide</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/">NCBI</a></td>
<td>Large subunit ribosomal RNA sequences for eukaryotic sequences</td>
</tr>
<tr class="odd">
<td><strong>LSU_prokaryote_rRNA</strong></td>
<td>nucleotide</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/">NCBI</a></td>
<td>Large subunit ribosomal RNA sequences for prokaryotic sequences</td>
</tr>
<tr class="even">
<td><strong>mito</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>NCBI Genomic Mitochondrial Reference Sequences</td>
</tr>
<tr class="odd">
<td><strong>mouse_genome</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/documents/blastdb.html">NCBI</a></td>
<td>Current RefSeq mouse genome assembly with various database masking</td>
</tr>
<tr class="even">
<td><strong>nr</strong></td>
<td>protein</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Non-redundant protein sequences from GenPept, Swissprot, PIR, PDF, PDB, and NCBI RefSe</td>
</tr>
<tr class="odd">
<td><strong>nt</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Partially non-redundant nucleotide sequences from all traditional divisions of GenBank, EMBL, and DDBJ</td>
</tr>
<tr class="even">
<td><strong>pataa</strong></td>
<td>protein</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Patent protein sequences</td>
</tr>
<tr class="odd">
<td><strong>patnt</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Patent nucleotide sequences. Both patent databases are directly from the USPTO, or from the EPO/JPO via EMBL/DDBJ</td>
</tr>
<tr class="even">
<td><strong>pdbaa</strong></td>
<td>protein</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Sequences for the protein structure from the Protein Data Bank</td>
</tr>
<tr class="odd">
<td><strong>pdbnt</strong></td>
<td>nucleitide</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Sequences for the nucleotide structure from the Protein Data Bank. They are NOT the protein coding sequences for the corresponding pdbaa entries.</td>
</tr>
<tr class="even">
<td><strong>ref_euk_rep_genomes</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/documents/blastdb.html">NCBI</a></td>
<td>Refseq Representative Eukaryotic genomes (1000+ organisms)</td>
</tr>
<tr class="odd">
<td><strong>ref_prok_rep_genomes</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/documents/blastdb.html">NCBI</a></td>
<td>Refseq Representative Prokaryotic genomes (5700+ organisms)</td>
</tr>
<tr class="even">
<td><strong>ref_viroid_rep_genomes</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/documents/blastdb.html">NCBI</a></td>
<td>Refseq Representative Viroid genomes (46 organisms)</td>
</tr>
<tr class="odd">
<td><strong>ref_viruses_rep_genomes</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/blast/documents/blastdb.html">NCBI</a></td>
<td>Refseq Representative Virus genomes (9000+ organisms)</td>
</tr>
<tr class="even">
<td><strong>refseq_protein</strong></td>
<td>protein</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>NCBI protein reference sequences</td>
</tr>
<tr class="odd">
<td><strong>refseq_rna</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>NCBI Transcript reference sequences</td>
</tr>
<tr class="even">
<td><strong>refseq_select_prot</strong></td>
<td>protein</td>
<td><a href="https://www.ncbi.nlm.nih.gov/refseq/refseq_select/">NCBI</a></td>
<td>NCBI RefSeq protein sequences from human, mouse, and prokaryotes, restricted to the RefSeq Select set of proteins. RefSeq Select includes one representative protein per protein-coding gene for human and mouse, and RefSeq proteins annotated on reference and representative genomes for prokaryotes</td>
</tr>
<tr class="odd">
<td><strong>refseq_select_rna</strong></td>
<td>nucleotide</td>
<td><a href="https://www.ncbi.nlm.nih.gov/refseq/refseq_select/">NCBI</a></td>
<td>NCBI RefSeq transcript sequences from human and mouse, restricted to the RefSeq Select set with one representative transcript per protein-coding gene</td>
</tr>
<tr class="even">
<td><strong>SSU_eukaryote_rRNA</strong></td>
<td>nucleotide</td>
<td><a href="https://ncbiinsights.ncbi.nlm.nih.gov/2020/02/21/rrna-databases/">NCBI</a></td>
<td>Small subunit ribosomal RNA sequences for eukaryotic sequences</td>
</tr>
<tr class="odd">
<td><strong>swissprot</strong></td>
<td>protein</td>
<td><a href="ftp://ftp.ncbi.nih.gov/blast/db/README">NCBI</a></td>
<td>Swiss-Prot sequence database (last major update)</td>
</tr>
<tr class="even">
<td><strong>tsa_nr</strong></td>
<td>protein</td>
<td><a href="https://www.ncbi.nlm.nih.gov/genbank/tsa/">NCBI</a></td>
<td>Protein sequences from the Trascriptome Shotgun Assembly. Its entries are EXCLUDED from the nr database.</td>
</tr>
<tr class="odd">
<td><strong>tsa_nt</strong></td>
<td>nucleotide</td>
<td><a href="https://www.ncbi.nlm.nih.gov/genbank/tsa/">NCBI</a></td>
<td>A database with earlier non-project based Transcriptome Shotgun Assembly (TSA) entries. Project-based TSA entries are NOT included. Entries are EXCLUDED from the nt database.</td>
</tr>
<tr class="even">
<td><strong>uniprot_sprot</strong></td>
<td>protein</td>
<td><a href="http://www.uniprot.org/help/uniprotkb_sections">UniProt</a></td>
<td>Swiss-Prot high quality manually annotated and non-redundant protein sequence database</td>
</tr>
<tr class="odd">
<td><strong>uniprot_trembl</strong></td>
<td>protein</td>
<td><a href="http://www.uniprot.org/help/uniprotkb_sections">UniProt</a></td>
<td>TrEMBL high quality but unreviewed protein sequence database</td>
</tr>
<tr class="even">
<td><strong>uniprot_sptrembl</strong></td>
<td>protein</td>
<td></td>
<td><strong>uniprot_sprot</strong> and <strong>uniprot_trembl</strong> combined</td>
</tr>
<tr class="odd">
<td><strong>uniprot_all</strong></td>
<td>protein</td>
<td></td>
<td>alias for <strong>uniprot_sptrembl</strong></td>
</tr>
<tr class="even">
<td><strong>uniprot_all.fasta</strong></td>
<td>protein</td>
<td></td>
<td>alias for <strong>uniprot_sptrembl</strong></td>
</tr>
<tr class="odd">
<td><strong>uniprot_sprot_varsplic</strong></td>
<td>protein</td>
<td><a href="http://www.uniprot.org/help/canonical_and_isoforms">UniProt</a></td>
<td>UniProt canonical and isoform sequences (see link)</td>
</tr>
<tr class="even">
<td><strong>uniprot_uniref50</strong></td>
<td>protein</td>
<td><a href="http://www.uniprot.org/uniref/">UniProt</a></td>
<td>Clustered sets of 50%-similar protein sequences (see link)</td>
</tr>
<tr class="odd">
<td><strong>uniprot_uniref90</strong></td>
<td>protein</td>
<td><a href="http://www.uniprot.org/uniref/">UniProt</a></td>
<td>Clustered sets of 90%-similar protein sequences (see link)</td>
</tr>
<tr class="even">
<td><strong>uniprot_uniref100</strong></td>
<td>protein</td>
<td><a href="http://www.uniprot.org/uniref/">UniProt</a></td>
<td>Clustered sets of identical protein sequences (see link)</td>
</tr>
<tr class="odd">
<td><strong>UniVec</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/pub/UniVec/README.uv">UniVec</a></td>
<td>Sequences commonly attached to cDNA/genomic DNA during the cloning process</td>
</tr>
<tr class="even">
<td><strong>UniVec_Core</strong></td>
<td>nucleotide</td>
<td><a href="ftp://ftp.ncbi.nlm.nih.gov/pub/UniVec/README.uv">UniVec</a></td>
<td>A subset of UniVec chosen to minimise false positives</td>
</tr>
</tbody>
</table>
<p>Additionally, <code>taxdb.btd</code> and <code>taxdb.bti</code> are downloaded, which provide <a href="ftp://ftp.ncbi.nih.gov/blast/db/README">additional taxonomy information for these databases</a>. Local copies of the NCBI Taxonomy databases are also available; <a href="http://www.uppmax.uu.se/resources/databases/ncbi-taxonomy-databases">further details are available on a separate page</a>.</p>
<p>For <code>UniVec</code> and <code>UniVec_Core</code>, Fasta-format files containing the vector sequences are also available with the given names (e.g., <code>/sw/data/uppnex/blast_databases/UniVec</code>), alongside the Blast-format databases built from the same Fasta files.</p>
<p>The exact times all databases were updated are provided by <em>database</em>.timestamp files located in the directory</p>
<h2 id="databases-are-available-automatically-after-loading-any-blast-module">Databases are available automatically after loading any blast module</h2>
<p>When any of the <strong>blast</strong> modules is loaded, the <code>BLASTDB</code> environment variable is set to the location of the local database copies (<code>/sw/data/uppnex/blast_databases</code>). The various Blast tools can use this variable to find the locations of databases, so that only the name needs to be specified.</p>
<pre><code>module load bioinfo-tools blast/2.7.1+
blastp -db nr -query input.fasta</code></pre>
<p>After loading the <code>blast/2.7.1+</code> module, specifying <code>blastp -db nr</code> results in <code>blastp</code> searching the local copy of <code>nr</code>, because the <code>BLASTDB</code> environment variable is set when the module is loaded. Similarly, each of these would result in searching the local copy of the given database:</p>
<pre><code>blastp -db pdbaa ...
blastp -db uniprot_sprot ...
blastp -db uniprot_uniref90 ...
blastn -db nt ...
blastn -db refseq_genomic ...</code></pre>
<h2 id="wgs-and-sra-sequence-databases-are-not-included">WGS and SRA sequence databases are not included</h2>
<p>The NCBI <a href="https://www.ncbi.nlm.nih.gov/genbank/wgs/">Whole-Genome Shotgun</a> is not available locally. NCBI provides special versions of Blast and other tools that can be used to search the remote versions of <a href="https://www.ncbi.nlm.nih.gov/genbank/wgs/">WGS</a> and the <a href="https://www.ncbi.nlm.nih.gov/sra">Sequence Read Archive</a>.</p>
<p>These special blast versions and other tools are part of NCBI's <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc">SRA Tools</a>, which is available at Uppmax as the <strong>sratools</strong> module. We have also include <a href="ftp://ftp.ncbi.nlm.nih.gov/blast/WGS_TOOLS/README_BLASTWGS.txt">auxiliary NCBI scripts</a> in the <strong>sratools</strong> module to convert taxonomic IDs to WGS and SRA identifiers.</p>
<p>Note that NCBI's TSA database is available at UPPMAX, just use the database name <code>tsa_nr</code> or <code>tsa_nt</code>.</p>
