vep/111.0
========================

<http://>

Used under license:



Structure creating script (makeroom_vep_111.0.sh) moved to /sw/bioinfo/vep/makeroom_111.0.sh

LOG
---

    /home/bjornv/git/install-methods/makeroom.sh "-t" "vep" "-v" "111.0" "-f"
    ./makeroom_vep_111.0.sh

# Clone repo and checkout v${VERSION}
    cd $SRCDIR
    git clone https://github.com/Ensembl/ensembl-vep.git .
    git reset --hard release/${VERSION%.*}

# Load modules
    module load perl_modules/5.26.2
    module load gcc/12.3.0
    module load MariaDB/10.2.11

# Add data cache dir on /sw/data/
    export CACHEDIR=/sw/data/$TOOL/$VERSION
    mkdir -p $CACHEDIR

# Export paths, otherwise the install fails
    export PATH=$PREFIX/htslib:$PATH
    export PERL5LIB=$PREFIX:$PERL5LIB

# Install and answer yes to the questions. Also accept the local database install.
    perl INSTALL.pl -d $PREFIX -t -c $CACHEDIR


## Install plugins
# Install the Bio::DB::BigFile plugin
    mkdir $SRCDIR/BigFile
    cd $SRCDIR/BigFile
    wget https://github.com/ucscGenomeBrowser/kent/archive/v335_base.tar.gz
    tar xzf v335_base.tar.gz --strip-components=1

    export KENT_SRC=$SRCDIR/BigFile/src
    export MACHTYPE=$(uname -m)
    export CFLAGS="-fPIC"
    export MYSQLINC=`mysql_config --include | sed -e 's/^-I//g'`
    export MYSQLLIBS=`mysql_config --libs`

    cd $KENT_SRC/lib
    echo 'CFLAGS="-fPIC"' > ../inc/localEnvironment.mk

    make clean && make -j 8
    cd ../jkOwnLib
    make clean && make -j 8

    mkdir $SRCDIR/BigFile/cpanm
    export PERL5LIB=$PERL5LIB:$SRCDIR/BigFile/cpanm/lib/perl5
    cpanm -l $SRCDIR/BigFile/cpanm Bio::DB::BigFile
    cpanm -l $SRCDIR/BigFile/cpanm Try::Tiny
    #cpanm -l $SRCDIR/BigFile/cpanm Bio::DB::HTS::VCF

# Copy scripts to $PREFIX
    mkdir -p $PREFIX/scripts
    cp -av * $PREFIX/scripts/
    cd $PREFIX/scripts/

# Do the tests
    module load vep/$VERSION
    cd $PREFIX/scripts
    perl -Imodules t/AnnotationSource_File_BigWig.t

## UTRAnnotator:
    cd $PREFIX
    git clone https://github.com/Ensembl/VEP_plugins.git
    cd VEP_plugins
    git reset --hard release/${VERSION%.*}
# Download test data    
    cd $SRCDIR
    git clone https://github.com/Ensembl/UTRannotator.git
    cd UTRannotator
    git reset --hard  40e30db003d72435eb3
    cd test
    mkdir  $CACHEDIR/UTRAnnotator
    cp $SRCDIR/UTRannotator/uORF_5UTR_GRCh3* $CACHEDIR/UTRAnnotator 
    vep -v -i $SRCDIR/UTRannotator/test/test_grch38.vcf \
        --assembly GRCh38 \
        --cache --dir $VEP_CACHE \
        --plugin UTRAnnotator,file=$CACHEDIR/UTRAnnotator/uORF_5UTR_GRCh38_PUBLIC.txt \
        --fasta /crex/data/iGenomes/Homo_sapiens/GATK/GRCh38/Sequence/WholeGenomeFasta/Homo_sapiens_assembly38.fasta \
        --force_overwrite 

## SpliceAI
# Download iLLumina data from https://basespace.illumina.com/projects/66029966. Yes... you have to create an account.
# The files are under genome_scores_v1.3. For GRCh37 the files are called spliceai_scores.raw.snv.hg19.vcf.gz  and spliceai_scores.raw.indel.hg19.vcf.gz
# I downloaded using thinlinc

    # Copy from older version
    ln -s /sw/data/vep/109.3/SpliceAI /sw/data/vep/${VERSION}/SpliceAI
    vep -v -i  $SRCDIR/UTRannotator/test/test_grch38.vcf \
        --plugin SpliceAI,snv=$CACHEDIR/SpliceAI/spliceai_scores.raw.snv.hg38.vcf.gz,indel=$CACHEDIR/SpliceAI/spliceai_scores.raw.indel.hg38.vcf.gz \
        --force_overwrite


## CADD
    ln -s /sw/data/vep/109.3/CADD /sw/data/vep/${VERSION}/CADD

    vep -i $VEP_TEST/homo_sapiens_GRCh38.vcf --plugin CADD,$VEP_CACHE/CADD/GRCh38/whole_genome_SNVs.tsv.gz,$VEP_CACHE/CADD/GRCh38/gnomad.genomes.r3.0.indel.tsv.gz -v --force_overwrite


# 2024-02-13
## BayesDel
mkdir $CACHEDIR/BayesDel
cd $CACHEDIR/BayesDel
# Download files from the google drive. https://drive.google.com/drive/folders/1K4LI6ZSsUGBhHoChUtegC8bgCt7hbQlA
# Extract
for file in *; do tar xfvz $file; done
# Remove tbi files
rm $CACHEDIR/BayesDel/*/*.tbi
# Expand all the chromosomes
find -iname "*.gz" -exec gunzip {} \;

cd $CACHEDIR/BayesDel/BayesDel_170824_addAF
for f in BayesDel_170824_addAF_chr*; do grep -v "^#" $f >> BayesDel_170824_addAF.txt; done
cat BayesDel_170824_addAF.txt | sort -k1,1 -k2,2n > BayesDel_170824_addAF_sorted.txt
grep "^#" BayesDel_170824_addAF_chr1 > BayesDel_170824_addAF_all_scores.txt
cat BayesDel_170824_addAF_sorted.txt >> BayesDel_170824_addAF_all_scores.txt
bgzip BayesDel_170824_addAF_all_scores.txt
tabix -s 1 -b 2 -e 2 BayesDel_170824_addAF_all_scores.txt.gz


cd $CACHEDIR/BayesDel/BayesDel_170824_noAF
for f in BayesDel_170824_noAF_chr*; do grep -v "^#" $f >> BayesDel_170824_noAF.txt; done
cat BayesDel_170824_noAF.txt | sort -k1,1 -k2,2n > BayesDel_170824_noAF_sorted.txt
grep "^#" BayesDel_170824_noAF_chr1 > BayesDel_170824_noAF_all_scores.txt
cat BayesDel_170824_noAF_sorted.txt >> BayesDel_170824_noAF_all_scores.txt
bgzip BayesDel_170824_noAF_all_scores.txt
tabix -s 1 -b 2 -e 2 BayesDel_170824_noAF_all_scores.txt.gz

# test BayesDel
vep -i $VEP_TEST/homo_sapiens_GRCh38.vcf --plugin BayesDel,file=/crex/data/vep/111.0/BayesDel/BayesDel_170824_addAF/BayesDel_170824_addAF_all_scores.txt.gz


## REVEL
mkdir $VEP_CACHE/REVEL
cd $VEP_CACHE/REVEL
wget https://rothsj06.dmz.hpc.mssm.edu/revel-v1.3_all_chromosomes.zip
unzip revel-v1.3_all_chromosomes.zip
cat revel_with_transcript_ids | tr "," "\t" > tabbed_revel.tsv
sed '1s/.*/#&/' tabbed_revel.tsv > new_tabbed_revel.tsv
bgzip new_tabbed_revel.tsv

tabix -f -s 1 -b 2 -e 2 new_tabbed_revel.tsv.gz # for GRCh37:

zcat new_tabbed_revel.tsv.gz | head -n1 > h # for GRCh38:
zgrep -h -v ^#chr new_tabbed_revel.tsv.gz | awk '$3 != "." ' | sort -k1,1 -k3,3n - | cat h - | bgzip -c > new_tabbed_revel_grch38.tsv.gz
tabix -f -s 1 -b 3 -e 3 new_tabbed_revel_grch38.tsv.gz

vep -i $VEP_TEST/homo_sapiens_GRCh38.vcf --assembly GRCh38 --plugin REVEL,file=$VEP_CACHE/REVEL/new_tabbed_revel_grch38.tsv.gz --force_overwrite

## 2024-04-04 Alphamissense
    module load GoogleCloudSDK/455.0.0
    module load tabix/0.2.6

    mkdir $CACHEDIR/AlphaMissense
    cd $CACHEDIR/Alphamissense

    gsutil -m cp \
        "gs://dm_alphamissense/AlphaMissense_aa_substitutions.tsv.gz" \
        "gs://dm_alphamissense/AlphaMissense_gene_hg19.tsv.gz" \
        "gs://dm_alphamissense/AlphaMissense_gene_hg38.tsv.gz" \
        "gs://dm_alphamissense/AlphaMissense_hg19.tsv.gz" \
        "gs://dm_alphamissense/AlphaMissense_hg38.tsv.gz" \
        "gs://dm_alphamissense/AlphaMissense_isoforms_aa_substitutions.tsv.gz" \
        "gs://dm_alphamissense/AlphaMissense_isoforms_hg38.tsv.gz" \
        "gs://dm_alphamissense/README.pdf" \
        .

    tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg38.tsv.gz
    tabix -s 1 -b 2 -e 2 -f -S 1 AlphaMissense_hg19.tsv.gz

    # test
    module load vep/111.0
    ./vep -i $VEP_TEST/homo_sapiens_GRCh38.vcf --plugin AlphaMissense,file=$VEP_CACHE/AlphaMissense/AlphaMissense_hg38.tsv.gz










vep/110.1
========================

<http://>

Used under license:



Structure creating script (makeroom_vep_110.1.sh) moved to /sw/bioinfo/vep/makeroom_110.1.sh

LOG
---

    /home/bjornv/git/install-methods/makeroom.sh "-t" "vep" "-v" "110.1" "-f"
    ./makeroom_vep_110.1.sh


# Clone repo and checkout v${VERSION}
    cd $SRCDIR
    git clone https://github.com/Ensembl/ensembl-vep.git .
    git reset --hard release/${VERSION%.*}

# Load modules
    module load perl_modules/5.26.2

# Add data cache dir on /sw/data/
    CACHEDIR=/sw/data/$TOOL/$VERSION
    mkdir -p $CACHEDIR

# Export paths, otherwise the install fails
    export PATH=$PREFIX/htslib:$PATH
    export PERL5LIB=$PREFIX:$PERL5LIB

# Install and answer yes to the questions. Also accept the local database install.
    perl INSTALL.pl -d $PREFIX -t -c $CACHEDIR

# Many questions are asked during the install. Say yes to all. 
# After downloading the local cache files the error "ERROR: For technical reasons this installer is unable to install GRCh37 caches alongside others; please install them separate" appears. 
# The installer downloads the 3 homo_sapiens GRCh37 and it seems OK. I will continue. 

# Copy scripts to $PREFIX
    mkdir -p $PREFIX/scripts
    cp -av * $PREFIX/scripts/
    cd $PREFIX/scripts/
    rm -rf t

# Test vep. Run post_install first
    cd $SRCDIR/ensembl-vep
    vep -v -i /sw/bioinfo/vep/${VERSION}/rackham/scripts/examples/homo_sapiens_GRCh38.vcf --cache --dir $VEP_CACHE --force_overwrite  --assembly GRCh38


### Install plugins
## UTRAnnotator:
    cd $PREFIX
    git clone https://github.com/Ensembl/VEP_plugins.git
    cd VEP_plugins
    git reset --hard release/${VERSION%.*}
# Download test data    
    cd $SRCDIR
    git clone https://github.com/Ensembl/UTRannotator.git
    cd UTRannotator
    git reset --hard  40e30db003d72435eb3
    cd test
    mkdir  $CACHEDIR/UTRAnnotator
    cp $SRCDIR/UTRannotator/uORF_5UTR_GRCh3* $CACHEDIR/UTRAnnotator 
    vep -v -i $SRCDIR/UTRannotator/test/test_grch38.vcf \
        --assembly GRCh38 \
        --cache --dir $VEP_CACHE \
        --plugin UTRAnnotator,file=$CACHEDIR/UTRAnnotator/uORF_5UTR_GRCh38_PUBLIC.txt \
        --force_overwrite 

## SpliceAI
# Download iLLumina data from https://basespace.illumina.com/projects/66029966. Yes... you have to create an account.
# The files are under genome_scores_v1.3. For GRCh37 the files are called spliceai_scores.raw.snv.hg19.vcf.gz  and spliceai_scores.raw.indel.hg19.vcf.gz
# I downloaded using thinlinc

    # Copy from older version
    ln -s /sw/data/vep/109.3/SpliceAI /sw/data/vep/${VERSION}/SpliceAI
    vep -v -i  $SRCDIR/UTRannotator/test/test_grch38.vcf \
        --plugin SpliceAI,snv=$CACHEDIR/SpliceAI/spliceai_scores.raw.snv.hg38.vcf.gz,indel=$CACHEDIR/SpliceAI/spliceai_scores.raw.indel.hg38.vcf.gz \
        --force_overwrite


## CADD
    ln -s /sw/data/vep/109.3/CADD /sw/data/vep/${VERSION}/CADD

    vep -i $VEP_TEST/homo_sapiens_GRCh38.vcf --plugin CADD,$VEP_CACHE/CADD/GRCh38/whole_genome_SNVs.tsv.gz,$VEP_CACHE/CADD/GRCh38/gnomad.genomes.r3.0.indel.tsv.gz -v --force_overwrite




vep/109.3
========================

<http://>

Used under license:



Structure creating script (makeroom_vep_109.3.sh) moved to /sw/bioinfo/vep/makeroom_109.3.sh

LOG
---

    /home/bjornv/git/install-methods/makeroom.sh "-t" "vep" "-v" "109.3" "-s" "misc" "-f"
    ./makeroom_vep_109.3.sh

    TOOL='vep'
    VERSION='109.3'

# Clone repo and checkout v109.3
    cd $SRCDIR
    git clone https://github.com/Ensembl/ensembl-vep.git
    cd ensembl-vep
    git checkout release/${VERSION}

# Load modules
    module load perl_modules/5.26.2

# Add data cache dir on /sw/data/
    CACHEDIR=/sw/data/$TOOL/$VERSION
    mkdir -p $CACHEDIR

# Export paths, otherwise the install fails
    export PATH=$PREFIX/htslib:$PATH
    export PERL5LIB=$PREFIX:$PERL5LIB

# Install and answer yes to the questions. Also accept the local database install.
    perl INSTALL.pl -d $PREFIX -t -c $CACHEDIR

# Many questions are asked during the install. Say yes to all. 
# After downloading the local cache files the error "ERROR: For technical reasons this installer is unable to install GRCh37 caches alongside others; please install them separate" appears. 
# The installer downloads the 3 homo_sapiens GRCh37 and it seems OK. I will continue. 

# Copy scripts to $PREFIX
    mkdir -p $PREFIX/scripts
    cp -av * $PREFIX/scripts/
    cd $PREFIX/scripts/
    rm -rf t

# Replace /usr/bin/perl with /usr/bin/env perl
    cd $PREFIX
    sed -r -i.bak 's/\#\!\/usr\/bin\/perl.*\s/\#\!\/usr\/bin\/env perl /' $(grep -rl --include=*.pl \#\!/usr/bin/perl)
    
# Test vep. Run post_install first
    cd $SRCDIR/ensembl-vep
    vep -v -i examples/homo_sapiens_GRCh38.vcf --cache --dir $VEP_CACHE --force_overwrite  --assembly GRCh38


### Install plugins
## UTRAnnotator:
    cd $PRERIX
    git clone https://github.com/Ensembl/VEP_plugins.git
    cd VEP_plugins
    git checkout release/109
# Download test data    
    cd $SRCDIR
    git clone https://github.com/Ensembl/UTRannotator.git
    cd UTRannotator
    git checkout 40e30db003d72435eb3
    cd test
    mkdir  $CACHEDIR/UTRAnnotator
    cp $SRCDIR/UTRannotator/uORF_5UTR_GRCh3* $CACHEDIR/UTRAnnotator 
    vep -v -i $SRCDIR/UTRannotator/test/test_grch38.vcf \
        --assembly GRCh38 \
        --cache --dir $VEP_CACHE \
        --plugin UTRAnnotator,file=$CACHEDIR/UTRAnnotator/uORF_5UTR_GRCh38_PUBLIC.txt \
        --force_overwrite 

## SpliceAI
# Download iLLumina data from https://basespace.illumina.com/projects/66029966. Yes... you have to create an account.
# The files are under genome_scores_v1.3. For GRCh37 the files are called spliceai_scores.raw.snv.hg19.vcf.gz  and spliceai_scores.raw.indel.hg19.vcf.gz
# I downloaded using thinlinc
    mkdir /sw/data/$TOOL/$VERSION/SpliceAI
    cd /sw/data/$TOOL/$VERSION/SpliceAI
    for file in *.vcf.gz; do tabix -p vcf $file -f & done
    vep -v -i  $SRCDIR/UTRannotator/test/test_grch38.vcf \
        --plugin SpliceAI,snv=$CACHEDIR/SpliceAI/spliceai_scores.raw.snv.hg38.vcf.gz,indel=$CACHEDIR/SpliceAI/spliceai_scores.raw.indel.hg38.vcf.gz \
        --force_overwrite


## CADD
    mkdir -p $CACHEDIR/CADD/{GRCh38,GRCh37}
    cd $CACHEDIR/CADD/GRCh38
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh38/whole_genome_SNVs.tsv.gz -b
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh38/whole_genome_SNVs.tsv.gz.tbi -b
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh38/gnomad.genomes.r3.0.indel.tsv.gz -b
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh38/gnomad.genomes.r3.0.indel.tsv.gz.tbi -b


    cd ../GRCh37
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh37/whole_genome_SNVs.tsv.gz -b
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh37/whole_genome_SNVs.tsv.gz.tbi -b
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh37/InDels.tsv.gz -b
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh37/InDels.tsv.gz.tbi -b
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh37/gnomad.genomes.r2.1.1.indel.tsv.gz -b 
    wget https://kircherlab.bihealth.org/download/CADD/v1.6/GRCh37/gnomad.genomes.r2.1.1.indel.tsv.gz.tbi -b

    vep -i $VEP_TEST/homo_sapiens_GRCh38.vcf \
        --plugin CADD,$VEP_CACHE/CADD/GRCh38/whole_genome_SNVs.tsv.gz,$VEP_CACHE/CADD/GRCh38/gnomad.genomes.r3.0.indel.tsv.gz \
        -v



vep/107
========================

<https://github.com/Ensembl/ensembl-vep>

Used under license:
Apache 2.0


Structure creating script (makeroom_vep_107.sh) moved to /sw/bioinfo/vep/makeroom_107.sh

LOG
---

    /home/bjornv/git/install-methods/makeroom.sh "-t" "vep" "-v" "107" "-l" "Apache 2.0" "-w" "https://github.com/Ensembl/ensembl-vep" "-s" "misc" "-f"
    ./makeroom_vep_107.sh
    TOOL='vep'
    VERSION='107'

# Clone repo and checkout v107
    cd $SRCDIR
    git clone https://github.com/Ensembl/ensembl-vep.git
    cd ensembl-vep
    git checkout release/107.0

# Load modules
    module load perl/2.26.2
    module load perl_modules/5.26.2

# Add data cache dir on /sw/data/
    CACHEDIR=/sw/data/$TOOL/$VERSION
    mkdir -p $CACHEDIR

# Export paths, otherwise the install fails
    export PATH=$PREFIX/htslib:$PATH
    export PERL5LIB=$PREFIX:$PERL5LIB

# Install and answer yes to the questions. Also accept the local database install.
    perl INSTALL.pl -d $PREFIX -t -c $CACHEDIR

# Many questions are asked during the install. Say yes to all. 
# After downloading the local cache files the error "ERROR: For technical reasons this installer is unable to install GRCh37 caches alongside others; please install them separate" appears. 
# The installer downloads the 3 homo_sapiens GRCh37 and it seems OK. I will continue. 

# Copy scripts to $PREFIX
    mkdir -p $PREFIX/scripts
    cp -av * $PREFIX/scripts/
    cd $PREFIX/scripts/
    rm -rf t

# Replace /usr/bin/perl with /usr/bin/env perl
    cd $PREFIX
    sed -r -i.bak 's/\#\!\/usr\/bin\/perl.*$/\#\!\/usr\/bin\/env perl/' $(grep -rl --include=*.pl \#\!/usr/bin/perl)
    

# Test with
    cd $SRCDIR/ensembl-vep
    vep -i examples/homo_sapiens_GRCh38.vcf --cache --dir $VEP_CACHE --force_overwrite  --assembly GRCh38











vep/99
========================

<https://github.com/Ensembl/ensembl-vep>

Used under license:
Apache 2.0

Structure creating script (makeroom_vep_99.sh) made with makeroom.sh (Author: Jonas Söderberg) and moved to /sw/bioinfo/vep/makeroom_99.sh

makeroom.sh invoked with:
/home/jonass/uppmax/install-methods/makeroom.sh -t "vep" -v "99" -w "https://github.com/Ensembl/ensembl-vep" -d "Variant Effect Predictor predicts the functional effects of genomic variants." -l "Apache 2.0" -f"

LOG
---

    screen -R vep
    /home/jonass/uppmax/install-methods/makeroom.sh -t "vep" -v "99" -w "https://github.com/Ensembl/ensembl-vep" -d "Variant Effect Predictor predicts the functional effects of genomic variants." -l "Apache 2.0" -f"
    ./makeroom_vep_99.sh
    cd $VERSIONDIR/src
    git clone https://github.com/Ensembl/ensembl-vep.git
    cd ensembl-vep
    module load perl/5.24.1
    module load perl_modules/5.24.1
    module load htslib/1.8
    CACHEDIR=/sw/data/uppnex/$TOOL/$VERSION
    mkdir -p $CACHEDIR
    PFX=$VERSIONDIR/$CLUSTER
    export PATH=$PFX/htslib:$PATH
    export PERL5LIB=$PFX:$PERL5LIB
    ./INSTALL.pl -d $PFX -t -c $CACHEDIR

NOTE: running the INSTALL.pl script will generate a question:

    Have you
    1. added /sw/apps/bioinfo/vep/87/milou to your PERL5LIB environment variable?
    2. added /sw/apps/bioinfo/vep/87/milou/htslib to your PATH environment variable?
    (y/n)

The two `export` lines above do that, so answer `y`.

I did not follow the next instruction:
##If on rackham, answer `n` to all cache installation questions.

Copy over the scripts.  This copies over the whole git tree, which is fine.  It
then removes the large test subdirectory.  Ensure all can find the proper perl
via `#!/usr/bin/env perl` line to  and (unlike earlier VEP versions) do not
deactiveate the install script so users can install their own to their own
project directories.

    mkdir -p $PFX/scripts
    cp -av * $PFX/scripts/
    cd $PFX/scripts/
    rm -rf t
    head -n 1 *.pl filter_vep haplo variant_recoder vep

vep/109.3
========================

<http://>

Used under license:



Structure creating script (makeroom_vep_109.3.sh) moved to /sw/bioinfo/vep/makeroom_109.3.sh

LOG
---

    /home/bjornv/git/install-methods/makeroom.sh "-t" "vep" "-v" "109.3" "-s" "misc" "-f"
    ./makeroom_vep_109.3.sh
