<p>Uppmax maintains local copies of nf-core pipelines (released) (<a href="https://nf-co.re/pipelines" class="uri">https://nf-co.re/pipelines</a>). Note that:</p>
<pre><code>    * The local copies can be loaded using &quot;$ module load nf-core-pipelines&quot;
    * To run a pipeline with a revision &quot;nextflow run $NF_CORE_PIPELINES/pipeline_name/revsion/workflow ...&quot;
    * The pipelines are updated **every sunday at 00.00**</code></pre>
<p>The pipelines available are:</p>
<table style="width:100%;">
<colgroup>
<col style="width: 26%" />
<col style="width: 42%" />
<col style="width: 31%" />
</colgroup>
<thead>
<tr class="header">
<th><strong>Name</strong></th>
<th><strong>Releases</strong></th>
<th><strong>Notes</strong></th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>ampliseq</strong></td>
<td>1.0.0, 1.1.0, 1.1.1, 1.1.2, 1.1.3, 1.2.0, 2.0.0, 2.1.0</td>
<td>16S rRNA amplicon sequencing analysis workflow using QIIME2</td>
</tr>
<tr class="even">
<td><strong>atacseq</strong></td>
<td>1.0.0, 1.1.0, 1.2.0, 1.2.1</td>
<td>ATAC-seq peak-calling, QC and differential analysis pipeline</td>
</tr>
<tr class="odd">
<td><strong>bacass</strong></td>
<td>1.0.0, 1.1.0, 1.1.1, 2.0.0</td>
<td>Simple bacterial assembly and annotation pipeline</td>
</tr>
<tr class="even">
<td><strong>bactmap</strong></td>
<td>1.0.0</td>
<td>A mapping-based pipeline for creating a phylogeny from bacterial whole genome sequences</td>
</tr>
<tr class="odd">
<td><strong>bcellmagic</strong></td>
<td>1.0.0, 1.1.0, 1.2.0, 2.0.0</td>
<td>B and T cell repertoire sequencing analysis pipeline using the Immcantation framework</td>
</tr>
<tr class="even">
<td><strong>cageseq</strong></td>
<td>1.0.0, 1.0.1, 1.0.2</td>
<td>CAGE-sequencing analysis pipeline with trimming, alignment and counting of CAGE tags</td>
</tr>
<tr class="odd">
<td><strong>chipseq</strong></td>
<td>1.0.0, 1.1.0, 1.2.0, 1.2.1, 1.2.2</td>
<td>ChIP-seq peak-calling, QC and differential analysis pipeline</td>
</tr>
<tr class="even">
<td><strong>clipseq</strong></td>
<td>1.0.0</td>
<td>CLIP sequencing analysis pipeline for QC, pre-mapping, genome mapping, UMI deduplication, and multiple peak-calling options</td>
</tr>
<tr class="odd">
<td><strong>coproid</strong></td>
<td>1.0, 1.1</td>
<td>Coprolite host Identification pipeline</td>
</tr>
<tr class="even">
<td><strong>deepvariant</strong></td>
<td>1.0</td>
<td>Googles DeepVariant variant caller as a Nextflow pipeline</td>
</tr>
<tr class="odd">
<td><strong>diaproteomics</strong></td>
<td>1.0.0 ,1.1.0, 1.2.0, 1.2.1, 1.2.2, 1.2.3 1.2.4</td>
<td>Automated quantitative analysis of DIA proteomics mass spectrometry measurements</td>
</tr>
<tr class="even">
<td><strong>dualrnaseq</strong></td>
<td>1.0.0</td>
<td>Analysis of Dual RNA-seq data - an experimental method for interrogating host-pathogen interactions through simultaneous RNA-seq</td>
</tr>
<tr class="odd">
<td><strong>eager</strong></td>
<td>2.0.0, 2.0.1, 2.0.2, 2.0.3, 2.0.4, 2.0.5, 2.0.6, 2.0.7, 2.1.0, 2.2.0, 2.2.1, 2.2.2, 2.3.0, 2.3.1, 2.3.2, 2.3.3, 2.3.4, 2.3.5, 2.4.0</td>
<td>A fully reproducible and state-of-the-art ancient DNA analysis pipeline</td>
</tr>
<tr class="even">
<td><strong>epitopeprediction</strong></td>
<td>1.0.0, 1.1.0</td>
<td>A bioinformatics best-practice analysis pipeline for epitope prediction and annotation</td>
</tr>
<tr class="odd">
<td><strong>fetchngs</strong></td>
<td>1.0, 1.1, 1.2, 1.3</td>
<td>Pipeline to fetch metadata and raw FastQ files from public databases</td>
</tr>
<tr class="even">
<td><strong>hic</strong></td>
<td>1.0.0, 1.1.0, 1.2.0, 1.2.1, 1.2.2, 1.3.0</td>
<td>Analysis of Chromosome Conformation Capture data (Hi-C)</td>
</tr>
<tr class="odd">
<td><strong>hlatyping</strong></td>
<td>1.0.0, 1.1.0, 1.1.1, 1.1.2, 1.1.3, 1.1.4, 1.1.5, 1.2.0</td>
<td>Precision HLA typing from next-generation sequencing data</td>
</tr>
<tr class="even">
<td><strong>imcyto</strong></td>
<td>1.0.0</td>
<td>Image Mass Cytometry analysis pipeline</td>
</tr>
<tr class="odd">
<td><strong>mag</strong></td>
<td>1.0.0, 1.1.0, 1.1.1, 1.1.2, 1.2.0, 2.0.0, 2.1.0</td>
<td>Assembly and binning of metagenomes</td>
</tr>
<tr class="even">
<td><strong>metaboigniter</strong></td>
<td>1.0.0, 1.0.1</td>
<td>Pre-processing of mass spectrometry-based metabolomics data with quantification and identification based on MS1 and MS2 data</td>
</tr>
<tr class="odd">
<td><strong>methylseq</strong></td>
<td>1.0, 1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.6.1</td>
<td>Methylation (Bisulfite-Sequencing) analysis pipeline using Bismark or bwa-meth + MethylDackel</td>
</tr>
<tr class="even">
<td><strong>mhcquant</strong></td>
<td>1.0.0, 1.1.0, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.2.4, 1.2.5, 1.2.6, 1.3, 1.4, 1.5, 1.5.1, 1.6.0, 2.0.0</td>
<td>Identify and quantify MHC eluted peptides from mass spectrometry raw data</td>
</tr>
<tr class="odd">
<td><strong>nanoseq</strong></td>
<td>1.0.0, 1.1.0</td>
<td>Nanopore demultiplexing, QC and alignment pipeline</td>
</tr>
<tr class="even">
<td><strong>nascent</strong></td>
<td>1.0</td>
<td>Nascent Transcription Processing Pipeline</td>
</tr>
<tr class="odd">
<td><strong>neutronstar</strong></td>
<td>1.0.0</td>
<td>De novo assembly pipeline for 10X linked-reads using Supernova</td>
</tr>
<tr class="even">
<td><strong>pgdb</strong></td>
<td>1.0.0</td>
<td>The ProteoGenomics database generation workflow creates different protein databases for ProteoGenomics data analysis</td>
</tr>
<tr class="odd">
<td><strong>proteomicslfq</strong></td>
<td>1.0.0</td>
<td>Proteomics label-free quantification (LFQ) analysis pipeline</td>
</tr>
<tr class="even">
<td><strong>rnafusion</strong></td>
<td>1.0, 1.0.1, 1.0.2, 1.1.0, 1.2.0</td>
<td>RNA-seq analysis pipeline for detection gene-fusions</td>
</tr>
<tr class="odd">
<td><strong>rnaseq</strong></td>
<td>1.0, 1.1, 1.2, 1.3, 1.4, 1.4.1, 1.4.2, 2.0, 3.0, 3.1, 3.2, 3.3</td>
<td>RNA sequencing analysis pipeline using STAR, RSEM, HISAT2 or Salmon with gene/isoform counts and extensive quality control</td>
</tr>
<tr class="even">
<td><strong>sarek</strong></td>
<td>2.5, 2.5.1, 2.5.2, 2.6, 2.6.1, 2.7, 2.7.1</td>
<td>Analysis pipeline to detect germline or somatic variants (pre-processing, variant calling and annotation) from WGS / targeted sequencing</td>
</tr>
<tr class="odd">
<td><strong>scrnaseq</strong></td>
<td>1.0.0, 1.1.0</td>
<td>A single-cell RNAseq pipeline for 10X genomics data</td>
</tr>
<tr class="even">
<td><strong>slamseq</strong></td>
<td>1.0.0</td>
<td>SLAMSeq processing and analysis pipeline</td>
</tr>
<tr class="odd">
<td><strong>smrnaseq</strong></td>
<td>1.0.0, 1.1.0</td>
<td>A small-RNA sequencing analysis pipeline</td>
</tr>
<tr class="even">
<td><strong>viralrecon</strong></td>
<td>1.0.0, 1.1.0, 2.0, 2.1, 2.2</td>
<td>Assembly and intrahost/low-frequency variant calling for viral samples</td>
</tr>
</tbody>
</table>
