<h1>Databases for the DIAMOND Protein Aligner</h1>
<p>The <a href="http://www.nature.com/nmeth/journal/v12/n1/full/nmeth.3176.html">iDIAMOND protein aligner</a> is a recent tool offering much faster (100× to 1000× faster than Blast) alignment of protein sequences against reference databases. On UPPMAX, DIAMOND is available by loading the <code>diamond</code> module, the most recent installed version of which which as of this writing is <code>diamond/2.0.14</code>.</p>
<p>As for <a href="https://www.uppmax.uu.se/blast-databases">BLAST databases</a>, UPPMAX provides several pre-built databases suitable for direct usage with the <code>--db</code> flag to <code>diamond</code>, as well as runs <code>diamond prepdb</code> on each of its downloaded BLAST protein databases whenever they are installed. The BLAST databases are updated according to the schedule given on their webpage. The <code>diamond</code>-format NCBI protein databases are updated once a month.</p>
<p>For each of the databases listed below, the method of versioning is indicated. To determine the version at UPPMAX, check the path given below after removing the database name from the last position; <code>latest</code> is a symbolic link that points to a directory with a name equivalent to the version of the most recent update. Old database versions will be removed after updates, so please use <code>latest</code> rather than directly addressing a database version.</p>
<p>Each of the database locations below is also available in the indicated environment variable set when any version of the diamond module is loaded. These are simple to use, for example to search nr:</p>
<pre><code>diamond --db $DIAMOND_NR ...</code></pre>
<h2>NCBI BLAST Protein Databases</h2>
<p>Whenever the BLAST databases are updated and installed, <code>diamond prepdb</code> is run on each of the protein-format databases so that they can be searched directly by <code>diamond</code>. See the BLAST databases webpage for a description of these.</p>
<p>To search any of them using <code>diamond</code>, load the <strong>blast_databases/latest</strong> module. This defines the environment variable <code>BLASTDB</code>, which contains the directory holding these databases. Once this module is loaded, you can run <code>diamond</code> on any of the protein databases. For example:</p>
<pre><code>diamond --db $BLASTDB/nr ...

diamond --db $BLASTDB/cdd_delta ...

diamond --db $BLASTDB/swissprot ...

diamond --db $BLASTDB/pdbaa ...</code></pre>
<p>According to DIAMOND's developer, these are faster to load than DIAMOND's own <code>.dmnd</code>-format databases. So, you may want to load the <code>blast_databases/latest</code> data module and use <code>--db $BLASTDB/nr</code> for your NCBI nr searches, for example, instead of <code>--db $DIAMOND_NR</code>.</p>
<h2>Diamond-format NCBI Protein Databases</h2>
<p>Downloaded from <a href="ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA" class="uri">ftp://ftp.ncbi.nlm.nih.gov/blast/db/FASTA</a>. These are updated frequently at NCBI, so they are versioned here by the monthly download date. There is no longer a separate FASTA version of <strong>env_nr</strong>, so its Blast database is downloaded from <a href="ftp://ftp.ncbi.nlm.nih.gov/blast/db" class="uri">ftp://ftp.ncbi.nlm.nih.gov/blast/db</a> and FASTA sequences are extracted using <code>blastdbcmd -entry all</code> from module <code>blast/2.12.0+</code>.</p>
<table>
<colgroup>
<col style="width: 16%" />
<col style="width: 48%" />
<col style="width: 34%" />
</colgroup>
<thead>
<tr class="header">
<th>Database</th>
<th>Environment variable for <code>diamond --db</code></th>
<th>UPPMAX path</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>nr</strong></td>
<td>DIAMOND_NR</td>
<td>/sw/data/diamond_databases/Blast/latest/nr</td>
</tr>
<tr class="even">
<td><strong>env_nr</strong></td>
<td>DIAMOND_ENV_NR</td>
<td>/sw/data/diamond_databases/Blast/latest/env_nr</td>
</tr>
<tr class="odd">
<td><strong>swissprot</strong></td>
<td>DIAMOND_SWISSPROT</td>
<td>/sw/data/diamond_databases/Blast/latest/swissprot</td>
</tr>
<tr class="even">
<td><strong>pdbaa</strong></td>
<td>DIAMOND_PDBAA</td>
<td>/sw/data/diamond_databases/Blast/latest/pdbaa</td>
</tr>
</tbody>
</table>
<h2>NCBI RefSeq Proteins</h2>
<p>RefSeq protein databases are downloaded from <a href="ftp://ftp.ncbi.nlm.nih.gov/refseq/release/complete/" class="uri">ftp://ftp.ncbi.nlm.nih.gov/refseq/release/complete/</a>, with an update occurring if there is a new release as indicated by the contents of <a href="ftp://ftp.ncbi.nlm.nih.gov/refseq/release/RELEASE_NUMBER" class="uri">ftp://ftp.ncbi.nlm.nih.gov/refseq/release/RELEASE_NUMBER</a>.</p>
<table>
<colgroup>
<col style="width: 35%" />
<col style="width: 40%" />
<col style="width: 24%" />
</colgroup>
<thead>
<tr class="header">
<th>Database</th>
<th>Environment variable for <code>diamond --db</code></th>
<th>UPPMAX path</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>complete.nonredundant_protein.protein</strong></td>
<td>DIAMOND_REFSEQ_NONREDUNDANT</td>
<td>/sw/data/diamond_databases/RefSeq/latest/complete.nonredundant_protein.protein</td>
</tr>
<tr class="even">
<td><strong>complete.protein</strong></td>
<td>DIAMOND_REFSEQ</td>
<td>/sw/data/diamond_databases/RefSeq/latest/complete.protein</td>
</tr>
</tbody>
</table>
<h2>UniRef90</h2>
<p>The UniRef90 protein database is downloaded as Fasta from its UK mirror at <a href="ftp://ftp.expasy.org/databases/uniprot/current_release/uniref/uniref90/" class="uri">ftp://ftp.expasy.org/databases/uniprot/current_release/uniref/uniref90/</a>, with an update occurring if there is a new version as indicated by the <code>&lt;version&gt;</code> tag in the XML description available at <a href="ftp://ftp.expasy.org/databases/uniprot/current_release/uniref/uniref90/RELEASE.metalink" class="uri">ftp://ftp.expasy.org/databases/uniprot/current_release/uniref/uniref90/RELEASE.metalink</a>.</p>
<table>
<colgroup>
<col style="width: 16%" />
<col style="width: 48%" />
<col style="width: 34%" />
</colgroup>
<thead>
<tr class="header">
<th>Database</th>
<th>Environment variable for <code>diamond --db</code></th>
<th>UPPMAX path</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>uniref90</strong></td>
<td>DIAMOND_UNIREF90</td>
<td>/sw/data/diamond_databases/UniRef90/latest/uniref90</td>
</tr>
</tbody>
</table>
<h2>UniProt Reference Proteomes</h2>
<p>The UniProt Reference Proteomes protein database is downloaded as Fasta from its UK mirror at <a href="ftp://ftp.expasy.org/databases/uniprot/current_release/knowledgebase/reference_proteomes" class="uri">ftp://ftp.expasy.org/databases/uniprot/current_release/knowledgebase/reference_proteomes</a>, with an update occurring if there is a new version as indicated by the <code>&lt;version&gt;</code> tag in the XML description available at <a href="ftp://ftp.expasy.org/databases/uniprot/current_release/knowledgebase/reference_proteomes/RELEASE.metalink" class="uri">ftp://ftp.expasy.org/databases/uniprot/current_release/knowledgebase/reference_proteomes/RELEASE.metalink</a>. If there is a new release, then the file <code>Reference_Proteomes_RELEASE.tar.gz</code> is downladed, with <code>RELEASE</code> replaced by the release number. The <code>reference_proteomes.dmnd</code> database is created from this file using the protocol described after the table.</p>
<table>
<colgroup>
<col style="width: 32%" />
<col style="width: 39%" />
<col style="width: 28%" />
</colgroup>
<thead>
<tr class="header">
<th>Database</th>
<th>Environment variable for <code>diamond --db</code></th>
<th>UPPMAX path</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td><strong>UniProt Reference Proteomes</strong></td>
<td>DIAMOND_REFERENCE_PROTEOMES</td>
<td>/sw/data/diamond_databases/reference_proteomes/latest/reference_proteomes</td>
</tr>
</tbody>
</table>
<p>The <code>reference_proteomes.dmnd</code> database is created using the following protocol for the <a href="https://blobtoolkit.genomehubs.org/install/">BlobToolKit</a>. This uses UPPMAX's most recently downloaded NCBI taxonomy database for its taxonomic metadata.</p>
<pre><code>module load bioinfo-tools
module load diamond/2.0.14
module load ncbi_taxonomy/latest

# after downloading
tar xf Reference_Proteomes_RELEASE.tar.gz
touch reference_proteomes.fasta.gz
find . -mindepth 2 | grep &quot;fasta.gz&quot; | grep -v &#39;DNA&#39; | grep -v &#39;additional&#39; | xargs cat &gt;&gt; reference_proteomes.fasta.gz
printf &quot;accession\taccession.version\ttaxid\tgi\n&quot; &gt; reference_proteomes.taxid_map
zcat */*/*.idmapping.gz | grep &quot;NCBI_TaxID&quot; | awk &#39;{print $1 &quot;\t&quot; $1 &quot;\t&quot; $3 &quot;\t&quot; 0}&#39; &gt;&gt; reference_proteomes.taxid_map

diamond makedb --db reference_proteomes.dmnd --in reference_proteomes.fasta.gz --threads 10 --taxonmap reference_proteomes.taxid_map --taxonnames $NCBI_TAXONOMY_ROOT/names.dmp --taxonnodes $NCBI_TAXONOMY_ROOT/nodes.dmp</code></pre>
