NextPolish/1.3.1
================

<https://github.com/Nextomics/NextPolish>

Used under license:


Structure creating script (makeroom_NextPolish_1.3.1.sh) moved to /sw/bioinfo/NextPolish/makeroom_1.3.1.sh

Installed slurm-drmaa/1.1.2-slurm19.05.8 prior to this.

Load python/3.8.7, install psutil locally, and load python/3.8.7 and the slurm-drmaa module in the mf file.


LOG
---

    cd /sw/bioinfo
    TOOL=NextPolish
    VERSION=1.3.1
    makeroom.sh -f -c bioinfo -s assembly -t $TOOL -v $VERSION -w https://github.com/Nextomics/NextPolish -d "fix base errors (SNV/Indel) in the genome generated by noisy long reads, it can be used with short read data only or long read data only or a combination of both"
    ./makeroom_NextPolish_1.3.1.sh 
    cd NextPolish/
    source SOURCEME_NextPolish_1.3.1 
    cd $SRCDIR
    wget https://github.com/Nextomics/NextPolish/releases/download/v1.3.1/NextPolish.tgz
    tar xf NextPolish.tgz 
    rmdir $PREFIX
    mv NextPolish $PREFIX
    cd $PREFIX

    module load python/3.8.7
    pip install --prefix=$PREFIX psutil
    pip install --prefix=$PREFIX drmaa
    module load slurm-drmaa/1.1.2-slurm19.05.8
    make

It installs everything here, and finds them based on the location of the nextPolish script.

Run the test now.

    export PYTHONPATH=$PREFIX/lib/python3.8/site-packages
    ./nextPolish test_data/run.cfg

Seems to work.  The mf file should load python/3.8.7, slurm-drmaa/1.1.2-slurm19.05.8, prepend to PYTHONPATH and prepend $PREFIX to PATH.
